src/System.Runtime/tests/System/StringGetHashCodeTests.cs

```diff
@@ -20,17 +20,24 @@ public class StringGetHashCodeTests : RemoteExecutorTestBase
         [MemberData(nameof(GetHashCode_TestData))]
         public void GetHashCodeWithStringComparer_UseSameStringInTwoProcesses_ReturnsDifferentHashCodes(int getHashCodeIndex)
         {
-            Func<string, string, int> method = (parentHash, i) => { return int.Parse(parentHash) != s_GetHashCodes[int.Parse(i)]() ? SuccessExitCode : -1; };
+            Func<string, string, int> method = (parentHash, i) => int.Parse(parentHash) != s_GetHashCodes[int.Parse(i)]() ? SuccessExitCode : -1;
             int parentHashCode = s_GetHashCodes[getHashCodeIndex]();
             int exitCode, retry = 0;
             do
             {
                 // very small chance the child and parent hashcode are the same. To further reduce chance of collision we try up to 3 times
                 using (RemoteInvokeHandle handle = RemoteInvoke(method, parentHashCode.ToString(), getHashCodeIndex.ToString(), new RemoteInvokeOptions { CheckExitCode = false }))
                 {
-                    Process p = handle.Process;
-                    p.WaitForExit();
-                    exitCode = p.ExitCode;
+                    if (PlatformDetection.IsUap)
+                    {
+                        exitCode = handle.ExitCode;
+                    }
+                    else
+                    {
+                        Process p = handle.Process;
+                        p.WaitForExit();
```

Comment:
I don't understand why you need every test to know that uap has to have special treatment. Why can't handle.ExitCode do WaitForExit then return the value? Then tests dont need to do special things, which they will rarely remember to do

Reply:
process is null in uap, because of this I cannot use WaitForExit

Reply:
I know, if it's null you wouldn't wait for it /use it. As a rule if there has to be complexity it's best to  minimize who has to be aware of it

Reply:
What are you suggesting? I'm not following